Assets/Scripts/Navigation/Map.cs

```
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Map : MonoBehaviour {
+
+	public GameObject hexPrefab;
+
+	int width = 20;
+	int height = 20;
+
+	float xOffset = 0.96f;
+	float yOffset = 0.71f;
+
+	public string mapShape = "hex";
+
+	public void Start(){
+		BuildMap();
+	}
+
+	void Update(){
+
+	}
+
+	private void BuildMap(){
+		switch(mapShape.ToUpper()){
+			//Builds a hex shaped map
+			case "HEX"	: //Todo
+			case "RECT" : RectMap();
+						  break;
+			//Builds a Rect shaped Map
+			default		: RectMap();
+					 	  break;
+		}
+	}
+
+	private void RectMap()
+	{
+		for(int x = 0; x < width; x++){
+			for(int y =0; y< height; y++){
+
+				float xPos = x * xOffset;
+
+				
+				if(y % 2 == 1){
+					xPos += xOffset/2f;
+				}
+
+
+				GameObject hex_go = (GameObject)Instantiate(hexPrefab, new Vector3(xPos,y * yOffset,0), Quaternion.identity);
+				hex_go.name = ("Hex_" + x + "_"+ y);
+
+				hex_go.GetComponent<Hex>().x = x;
+				hex_go.GetComponent<Hex>().y = y;
+
+				hex_go.tag = "Water";
+
+				hex_go.transform.SetParent(this.transform);
+			}
+		}
+	}
+
+	private void HexMap(){
+		Debug.Log("hex");
```

Does unity have an equivalent to "throw"? Let's just throw an error here instead of outputting text